test/server/dom-shims.js

```diff
@@ -41,3 +42,16 @@ describe(`customElement with shadowDom`, function () {
     expect(elem.el.innerHTML).to.equal(`<style>color: blue;</style><div class="foo"><p>Hello</p></div>`);
   });
 });
+
+describe(`appending customElements to DOM`, function () {
+  it(`calls connectedCallback on customElements when using appendChild`, async function () {
+    const elem = document.createElement(`my-app`);
+    const connectedCallbackStub = sinon.stub(elem, `connectedCallback`);
+
+    document.createElement(`div`).appendChild(elem);
+
+    await nextAnimationFrame();
+    expect(connectedCallbackStub.called).to.be.true;
+    expect(elem.isConnected).to.be.true;
+  });
+});
```

Comment:
TODO server-side test case that uses contexts??

Reply:
will look into porting the existing test cases in `browser/`

Reply:
ported two basic test cases from `browser/` in https://github.com/mixpanel/panel/pull/110/commits/8ff7019626b452676af8dc5f97cd6d93f4dc85ae

the more advanced cases will need further shimming (things we already take care of in `analytics`) and I didn't want to scope creep this PR any further. lmk how you feel 🙏